### DISCUSSION TITLE
AttributeError: 'Document' object has no attribute 'set_oc_states'

### DISCUSSION THREAD

**Discussion by erand96** [pymupdf/PyMuPDF#1008] [Looking for help] [answered]
I am new to PyMuPDF and recently have been trying to alter OCGs with it. However, there are a handful of attributes that refuse to work when calling them. I cannot find a pattern between the attributes that do and do not work.

For example, I have the basic code below:

```
import fitz

doc = fitz.open('H:\\testo.pdf')

doc.set_oc_states()
```

But when I run the program, I get the error in the title: AttributeError: 'Document' object has no attribute 'set_oc_states'. 

I can run other attributes such as `get_oc_status()` without any issues.

Has anyone else encountered this issue before or have any possible causes of this? I'm relatively new to Python as well as this particular library.

**Comment by JorjMcKie** [comment, marked as answer]
This is a documentation issue. OC-Support is relatively new, so there was initial juggling with the method names.
Instead of `doc.set_oc_states`, please use `doc.set_layer`. It is the same thing.
I will update the documentation accordingly.

**Reply by erand96** [reply]
Great, thank you for the quick reply!
